src/compiler/checker.ts

```diff
@@ -3177,9 +3177,7 @@ namespace ts {
             if (includePatternInType) {
                 result.pattern = pattern;
             }
-            if (hasComputedProperties) {
-                result.flags |= TypeFlags.ObjectLiteralPatternWithComputedProperties;
-            }
+            result.inObjectLiteralPatternWithComputedProperties = hasComputedProperties;

```

Comment:
Hm, turns out I was wrong when I thought that `x.prop = undefined` is a no-op. Fixed.